### PR TITLE
Adding Weapon entity

### LINKED ISSUE #28: Getting weapon name and info
Since `player.weapon` returns a `BaseEntity`, how to get that weapon name (i didnt found it on the entity props)? Could we consider making a class to Weapon (like Player, Team, etc.) ?

### PR DESCRIPTION
**Pull request by pedrofornaza**
closes #28

### CODE REVIEW

File: entities/weapon.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  1: 'Desert Eagle',
+  2: 'Dual Berettas',
+  3: 'Five-SeveN',
+  4: 'Glock-18',
+  7: 'AK-47',
+  8: 'AUG',
+  9: 'AWP',
+  10: 'FAMAS',
+  11: 'G3SG1',
+  13: 'Galil AR', 14: 'M249',
```
Comment (saul) [inline entities/weapon.js]: New line

File: entities/weapon.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+/**
+ * Represents an in-game weapon (guns, grenades, knifes).
+ */
+class Weapon extends BaseEntity {
+  /**
+   * @returns {string} Name of the weapon
+   */
+  get name() {
+    var weaponId = this.getProp('DT_ScriptCreatedItem', 'm_iItemDefinitionIndex');
```
Comment (saul) [inline entities/weapon.js]: Can we add a new itemIndex property for this?

File: entities/weapon.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  /**
+   * @returns {Player|null} Previous Owner
+   */
+  get prevOwner() {
+    var userId = this.getProp('DT_WeaponCSBase', 'm_hPrevOwner'),
+        user = null;
+
+    if (user = this._demo.entities.getByUserId(userId)) {
```
Comment (saul) [inline entities/weapon.js]: hPrevOwner is an entity handle. Use getByHandle instead

**Comment by pedrofornaza** [comment]
About the new name property: do we have a "right" naming convention? Maybe i can just make some changes based on the itemName return like prefixing "weapon_", making it all lower caps and so on.

**Comment by saul** [comment]
Yes the proper naming convention is here: https://github.com/SteamDatabase/GameTracking-CSGO/blob/master/csgo/scripts/items/items_game.txt

Search for "weapon_deagle" and you'll find it (it's about half way down the file).

**Comment by pedrofornaza** [comment]
What do you think? I dont know the best way to structure data on javascript.

**Comment by saul** [comment]
Looks good, thanks. Can you rename name to className too? Also see my earlier comment on entity handle vs userid

**Comment by saul** [comment]
Thanks for this @pedrofornaza

**Pull request merged**